plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/explorer/QStatusBarLoggedOutActionGroup.kt

```diff
@@ -0,0 +1,69 @@
+
+package software.aws.toolkits.jetbrains.services.codewhisperer.explorer
+
+import com.intellij.openapi.actionSystem.ActionManager
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DefaultActionGroup
+import com.intellij.openapi.actionSystem.Separator
+import software.aws.toolkits.jetbrains.core.credentials.AwsBearerTokenConnection
+import software.aws.toolkits.jetbrains.core.credentials.ToolkitConnectionManager
+import software.aws.toolkits.jetbrains.core.credentials.actions.SsoLogoutAction
+import software.aws.toolkits.jetbrains.core.credentials.pinning.CodeWhispererConnection
+import software.aws.toolkits.jetbrains.core.utils.buildList
+import software.aws.toolkits.jetbrains.services.codewhisperer.actions.CodeWhispererConnectOnGithubAction
+import software.aws.toolkits.jetbrains.services.codewhisperer.actions.CodeWhispererLearnMoreAction
+import software.aws.toolkits.jetbrains.services.codewhisperer.actions.CodeWhispererProvideFeedbackAction
+import software.aws.toolkits.jetbrains.services.codewhisperer.actions.CodeWhispererShowSettingsAction
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.actions.ActionProvider
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.actions.Customize
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.actions.Learn
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.actions.OpenCodeReference
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.actions.Pause
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.actions.Resume
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.actions.buildActionListForConnectHelp
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.actions.buildActionListForInlineSuggestions
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.actions.buildActionListForOtherFeatures
+import software.aws.toolkits.resources.message
+
+class QStatusBarLoggedOutActionGroup : DefaultActionGroup() {
+    private val loggedInActionProvider = object : ActionProvider<AnAction> {
+        override val pause = Pause()
+        override val resume = Resume()
+        override val openCodeReference = OpenCodeReference()
+        override val customize = Customize()
+        override val learn = Learn()
+        override val openChatPanel = ActionManager.getInstance().getAction("q.openchat")
+        override val runScan = ActionManager.getInstance().getAction("codewhisperer.toolbar.security.scan")
+        override val stopScan = ActionManager.getInstance().getAction("codewhisperer.toolbar.security.stopscan")
+        override val sendFeedback = CodeWhispererProvideFeedbackAction()
+        override val connectOnGithub = CodeWhispererConnectOnGithubAction()
+        override val documentation = CodeWhispererLearnMoreAction()
+    }
```

Comment:
provider is the same across both groups?
i also dont think we need the provider anymore if we're getting rid of the explorer nodes

Reply:
It's addition to the existing code.
I can clean them up if not needed anymore.